### PR TITLE
feat: :lipstick: create reusable components

### PR DESCRIPTION
**Pull request by YafiAhsanH**
### Background: 
add custom form, button, text button, and text field.

### Additional Packages/Dependencies:
- none

### Changes:
- common/components/custom_button.dart
- common/components/custom_text_button.dart
- common/components/custom_text_field.dart
- common/components/form_component.dart

### How to Implement:
- call the classes with the required parameters

### CODE REVIEW

**Comment by YafiAhsanH** [comment]
@suryanirvana pls check

File: lib/page/page.dart [hunk 1]
```diff
@@ -1,54 +1,95 @@
 import 'package:flutter/material.dart';
 import 'package:flutter_bloc/flutter_bloc.dart';
+// import 'package:flutter_boilerplate/auth/bloc/register/register_cubit.dart';
```
Comment (suryanirvana) [inline lib/page/page.dart]: apus ini dong

Comment (suryanirvana) [reply lib/page/page.dart]: dan sisanya

File: lib/main.dart [hunk 2]
```diff
@@ -53,7 +45,6 @@ class MyHomePage extends StatefulWidget {
 }
 
 class _MyHomePageState extends State<MyHomePage> {
-
   @override
   Widget build(BuildContext context) {
     // This method is rerun every time setState is called, for instance as done
```
Comment (suryanirvana) [inline lib/main.dart]: ini juga hehuehu

**Review by suryanirvana — APPROVED** [review]
looks-good-to-me

**Review by oliver-mf — APPROVED** [review]
all good 👍

**Pull request merged**